package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+    :func:`~MDAnalysis.coordinates.base.add_auxiliary`
+
+    Note
+    ----
+    The file is assumed to be of a size such that reading and storing the full
+    contents is practical.
+    """
+
+    format = "EDR"
+    _Auxstep = EDRStep
+
+    def __init__(self, filename: str, **kwargs):
+        self._auxdata = Path(filename).resolve()
+        self.auxdata = pyedr.edr_to_dict(filename)
+        self._n_steps = len(self.auxdata["Time"])
+        # attribute to communicate found energy terms to user
+        self.terms = [key for key in self.auxdata.keys()]
+        super(EDRReader, self).__init__(**kwargs)
+
+    def _read_next_step(self) -> EDRStep:
+        """Read next auxiliary step and update ``auxstep``.
+
+        Returns
+        -------
+        AuxStep object
+            Updated with the data for the new step.
+
+        Raises
+        ------
+        StopIteration
+            When end of auxiliary data set is reached.
+        """
+        auxstep = self.auxstep
+        new_step = self.step + 1
+        if new_step < self.n_steps:
+            auxstep._data = {term: self.auxdata[term][self.step + 1]
+                             for term in self.terms}
+            auxstep.step = new_step
+            return auxstep
+        else:
+            self.rewind()
+            raise StopIteration
+
+    def _go_to_step(self, i: int) -> EDRStep:
+        """ Move to and read i-th auxiliary step.
+
+        Parameters
+        ----------
+        i: int
+            Step number (0-indexed) to move to
+
+        Returns
+        -------
+        :class:`EDRStep`
+
+        Raises
+        ------
+        ValueError
+            If step index not in valid range.
+        """
+        if i >= self.n_steps or i < 0:
+            raise ValueError("Step index {0} is not valid for auxiliary "
+                             "(num. steps {1})".format(i, self.n_steps))
+        self.auxstep.step = i - 1
+        self.next()
+        return self.auxstep
+
+    def read_all_times(self) -> np.ndarray:
+        """ Get list of time at each step.
+
+        Returns
+        -------
+        NumPy array of float
+            Time at each step.
+        """
+        return self.auxdata[self.time_selector]
+
+    def return_data(self, data_selector: Union[str, List[str], None] = None) \
+            -> Dict[str, np.ndarray]:
+        """ Returns the auxiliary data contained in the :class:`EDRReader`.
+        Returns either all data or data specified as `data_selector` in form
+        of a str or a list of any of :attr:`EDRReader.terms`. `Time` is
+        always returned to allow easy plotting. """
+        if not data_selector or data_selector == "*":
```

Comment:
As mentioned above, I would just make the `None` case the one that users should use here.